### ISSUE TITLE
[GUI] no separator for thousands

### ISSUE THREAD

**Issue by kloaec** [wizardsardine/liana#962]
The balance display is really cool for the separations of the decimals and making sats visible.
We do not have a separation of the thousands though!

There are different ways to approach this, either the American way XX,XXX.XX or just spacing XX XXX.XX.
I prefer the space as a more "global" approach, given that some countries use the comma and other the dot for the decimals, I don't want them to be confused.

The same should be done for the millions.

![image](https://github.com/wizardsardine/liana/assets/15522963/4c1f34e0-03d6-4aea-8f00-5d1350261b36)

**Comment by pythcoiner** [comment]
I also in favor of space. (never test w/ such amount tbh :grinning:

**Comment by Nonnyjoe** [comment]
I'll Like to handle this issue

**Comment by darosior** [comment]
@Nonnyjoe welcome to the project. Sounds good, i'm assigning this issue to you.

**Comment by aaroncarlucci** [comment]
I looked into this a bit recently while attempting to onboard to the project. Unfortunately, haven't come up yet with a fix, but decided to leave some initial, possibly ignorant feedback here from someone picking up both Rust and the Liana stack.
* FYI - there were some very recent changes WRT to `bitcoin::Amount` formatting committed recently to the [rust-bitcoin](https://github.com/rust-bitcoin/rust-bitcoin/commit/014c4931be9edb530b1db15e14feeca5506cad6b) crate. I think that crate might support passing in a custom formatter to render amounts?
* That crate also makes use of overriding `fmt::Display` to define custom formatting. I played around locally with this, but it was over my head at the moment. Perhaps there is an opportunity to subclass `bitcoin::Amount` and define a custom formatter? Could be a good opportunity to refactor what appears to be a growing if/else statement in amount.rs?
* Another option might be to consider the [num_format](https://docs.rs/num-format/latest/num_format) crate and use "French" formatting, or define a custom format. Another dependency, though... and doesn't support floats, so render of BTC values might need to be split on the decimal and rendered separately?

Maybe I'll be able to come up with some code eventually, but hoping that sharing my research thus far helps somehow.

**Comment by pythcoiner** [comment]
Hi @aaroncarlucci welcome on board! 
I think we should not add any dependency for this.

In my mind the most straigtforward is to rework the logic in `amount.rs::split_digits()`:
- splitting `s` w/ '.'
- then chunk & join both sides (integer part should be done in reverse)
```rust
    let decimal = decimal
        .chars()
        .collect::<Vec<_>>()
        .chunks(3)
        .into_iter()
        .map(|c| c.iter().collect::<String>())
        .collect::<Vec<_>>()
        .join(" ");
```
- then join the produced strings

**Comment by edouardparis** [comment]
fixed by #1198
